File: frontend/src/api/models/Message.ts
import { User } from "./User";

export interface Message {
  type: MessageType;
  timestamp: string;
  content: string;
  user?: User;
  id?: number;
}

export enum MessageType {
  Message = "Message",
  Post = "Post",
  Status = "Status",

  // This is a client-side message type
  Date = "Date",
}

File: backend/src/main/kotlin/org/pulse/backend/entities/message/Message.kt
package org.pulse.backend.entities.message

import jakarta.persistence.*
import org.pulse.backend.entities.channel.Channel
import org.pulse.backend.entities.user.User
import org.pulse.backend.responses.MessageResponse
import java.sql.Timestamp

@Entity
class Message(
    val type: MessageType,
    val timestamp: Timestamp,
    @Lob
    var content: String,
    @ManyToOne
    val channel: Channel,
    @ManyToOne
    val user: User? = null,
    @OneToOne
    var comments: Channel? = null,
    @Id
    @GeneratedValue
    val id: Long? = null
) {
    fun toResponse(): MessageResponse {
        return MessageResponse(
            type,
            timestamp,
            content,
            user?.toResponse(),
            id!!
        )
    }
}